scylla-cdc/src/lib.rs

```diff
@@ -5,9 +5,426 @@ pub mod stream_generations;
 
 #[cfg(test)]
 mod tests {
-    #[test]
-    fn it_works() {
-        let result = 2 + 2;
-        assert_eq!(result, 4);
+    use crate::consumer::*;
+    use async_trait::async_trait;
+    use futures::StreamExt;
+    use scylla::frame::response::result::CqlValue;
+    use scylla::frame::value::{Value, ValueTooBig};
+    use scylla::prepared_statement::PreparedStatement;
+    use scylla::{Session, SessionBuilder};
+    use std::collections::{HashMap, VecDeque};
+    use std::hash::Hash;
+    use std::sync::Arc;
+    use tokio::sync::Mutex;
+
+    const MAX_PK_SIZE: usize = 16;
+
+    #[derive(Debug, Eq, PartialEq, Hash)]
+    enum PrimaryKeyValue {
+        // Name consistency with CqlValue from the driver is recommended.
+        Int(i32),
+        Text(String),
+        List(Vec<PrimaryKeyValue>),
+    }
+
+    impl Value for PrimaryKeyValue {
+        fn serialize(&self, buf: &mut Vec<u8>) -> Result<(), ValueTooBig> {
+            self.as_cql().serialize(buf)?;
+
+            Ok(())
+        }
+    }
+
+    impl PrimaryKeyValue {
+        pub fn from_cql(cql_val: &CqlValue) -> Option<PrimaryKeyValue> {
+            match cql_val {
+                CqlValue::Int(x) => Some(PrimaryKeyValue::Int(*x)),
+                CqlValue::Text(s) => Some(PrimaryKeyValue::Text(s.clone())),
+                _ => None,
+            }
+        }
+
+        pub fn as_cql(&self) -> CqlValue {
+            match self {
+                PrimaryKeyValue::Int(x) => CqlValue::Int(*x),
+                PrimaryKeyValue::Text(s) => CqlValue::Text(s.clone()),
+                PrimaryKeyValue::List(v) => {
```

Comment:
It's related with the other comment. `PrimaryKeyValue::List` was meant to indicate that the primary key is complex. Since it will not, I'll add the other conversion.